### ISSUE TITLE
Shouldn't assume there's a body

### ISSUE THREAD

**Issue by dlq84** [zerodine/angular-restclient#13]
I think this library is too strict and should not assume there's a body in case of error 4xx. it says SHOULD, and not MUST. Which by the definition of SHOULD in rfc2119 means it's only recommended.

Thoughts?

## 10.4 Client Error 4xx
"the server SHOULD include an entity containing an explanation of the error situation, and whether it is a temporary or permanent condition. These status codes are applicable to any request method. User agents SHOULD display any included entity to the user."

https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html

**Comment by philippspinnler** [comment]
You are absolutely right. I designed this library for a specific project in which there always was a body set in case of a 4xx error. I'm going to change it in 1.4.0 release or just feel free to create a pull request.
